frontend/tests/features/differentialExpression/differentialExpression.test.ts

```diff
@@ -0,0 +1,790 @@
+      );
+
+      // Get initial cell counts after selecting "lung"
+      const initialCellCountGroup1 = await page
+        .getByTestId(DIFFERENTIAL_EXPRESSION_CELL_GROUP_1_FILTER)
+        .getByTestId(DIFFERENTIAL_EXPRESSION_FILTER_CELL_COUNT)
+        .textContent();
+
+      // Switch organism dropdown to "Mus musculus"
+      const organismDropdown = page.getByTestId(
+        DIFFERENTIAL_EXPRESSION_ORGANISM_DROPDOWN
+      );
+      await organismDropdown.click();
+      await page.keyboard.press("ArrowDown");
+      await page.keyboard.press("Enter");
+      await waitForFiltersEndpoint(page);
+
+      // Ensure "lung" tag is no longer present
+      const lungTag = page
+        .getByTestId(DIFFERENTIAL_EXPRESSION_CELL_GROUP_1_FILTER)
+        .getByTestId(DIFFERENTIAL_EXPRESSION_FILTER_TAG_PRIMARY)
+        .filter({ hasText: "lung" });
+      await expect(lungTag).toHaveCount(0);
+
+      // Ensure cell counts changed
+      const updatedCellCountGroup1 = await page
+        .getByTestId(DIFFERENTIAL_EXPRESSION_CELL_GROUP_1_FILTER)
+        .getByTestId(DIFFERENTIAL_EXPRESSION_FILTER_CELL_COUNT)
+        .textContent();
+      expect(initialCellCountGroup1).not.toBe(updatedCellCountGroup1);
+    });
+
+    test("Find genes button only active when both filters populated", async ({
+      page,
+    }) => {
+      // Ensure "Find Genes" button is disabled initially
+      const findGenesButton = page.getByTestId(
+        DIFFERENTIAL_EXPRESSION_FIND_GENES_BUTTON
+      );
+      await expect(findGenesButton).toBeDisabled();
+
+      // Populate "lung" in cell group 1 filter tissue
+      const tissueFilterAutocompleteGroup1 = page
+        .getByTestId(DIFFERENTIAL_EXPRESSION_CELL_GROUP_1_FILTER)
+        .getByTestId(
+          `${DIFFERENTIAL_EXPRESSION_FILTER_AUTOCOMPLETE_PREFIX}Tissue`
+        );
+      await clickOnAutocompleteDropdownItem(
+        tissueFilterAutocompleteGroup1,
+        "lung"
+      );
+
+      // Ensure "Find Genes" button is still disabled
+      await expect(findGenesButton).toBeDisabled();
+
+      // Populate "blood" in cell group 2 filter tissue
+      const tissueFilterAutocompleteGroup2 = page
+        .getByTestId(DIFFERENTIAL_EXPRESSION_CELL_GROUP_2_FILTER)
+        .getByTestId(
+          `${DIFFERENTIAL_EXPRESSION_FILTER_AUTOCOMPLETE_PREFIX}Tissue`
+        );
+      await clickOnAutocompleteDropdownItem(
+        tissueFilterAutocompleteGroup2,
+        "blood"
+      );
+
+      // Ensure "Find Genes" button is enabled
+      await expect(findGenesButton).toBeEnabled();
+
+      // Delete "lung" in cell group 1 filter tissue
+      await tissueFilterAutocompleteGroup1
+        .getByTestId(DIFFERENTIAL_EXPRESSION_FILTER_TAG_PRIMARY)
+        .locator('[class*="CloseIcon"]')
+        .click();
+      await waitForFiltersEndpoint(page);
+      // Ensure "Find Genes" button is disabled again
+      await expect(findGenesButton).toBeDisabled();
+    });
+  });
+
+  describe("Results", () => {
```

Comment:
Because running a DE query takes a non-negligible amount of time, these E2E tests are all encapsulated as steps within a single test. That way, we only need to run the DE query one time. I explored using a mocked endpoint but it was too much work to set up the fixture.